price/pricing/config.go

```diff
@@ -72,43 +53,35 @@ func (cpd *ConfigProviderDB) Update(in Config) error {
 	if err != nil {
 		return err
 	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
 
-	conn, err := cpd.db.Connection()
-	if err != nil {
-		return err
-	}
-	defer conn.Release()
-
-	query := `INSERT INTO pricing_config(cfg) VALUES ($1);`
-	_, err = conn.Exec(context.Background(), query, cfgJSON)
+	err = cpd.db.Set(ctx, PricingConfigRedisKey, string(cfgJSON), 0).Err()
 	if err != nil {
 		return err
 	}
 
-	// invalidate the cache so that the pricing is updated on next Get.
-	cpd.cc.ValidUntil = time.Time{}
-
 	return nil
 }
 
 func (cpd *ConfigProviderDB) fetchConfig() (Config, error) {
-	conn, err := cpd.db.Connection()
-	if err != nil {
-		return Config{}, err
-	}
-	defer conn.Release()
-
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
-
-	res := Config{}
-	row := conn.QueryRow(ctx, "SELECT cfg FROM pricing_config order by id desc limit 1;")
-	err = row.Scan(&res)
+	val, err := cpd.db.Get(ctx, PricingConfigRedisKey).Result()
 	if err != nil {
-		return Config{}, err
+		if err.Error() == redis.Nil.Error() {
```

Comment:
```suggestion
		if errors.Is(err, redis.Nil) {
```

probably the same in other places too?